### PR TITLE
FINERACT-1981: Remove hardcoded MoneyHelper dependency during progressive loan

### PR DESCRIPTION
**Pull request by kulminsky**
Remove hardcoded MoneyHelper dependency during progressive loan

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java [hunk 1]
```diff
@@ -48,18 +49,25 @@
 @DependsOn("moneyHelper")
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java]: We can remove this now

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java [hunk 2]
```diff
@@ -48,18 +49,25 @@
...
 public final class ProgressiveEMICalculator implements EMICalculator {
 
+    private MathContext mc;
+
     private static final BigDecimal DIVISOR_100 = new BigDecimal("100");
     private static final BigDecimal ONE_WEEK_IN_DAYS = BigDecimal.valueOf(7);
 
+    @Override
+    public void setMathContext(final MathContext mc) {
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java]: We dont need this. EmiCalculator always work with `ProgressiveLoanInterestScheduleModel` and it already has the MathContext

File: fineract-progressive-loan/src/test/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleGeneratorTest.java [hunk 3]
```diff
@@ -87,7 +71,7 @@ void testGenerateLoanSchedule() {
 
         ScheduledDateGenerator mockScheduledDateGenerator = Mockito.mock(ScheduledDateGenerator.class);
         ProgressiveLoanScheduleGenerator generator = new ProgressiveLoanScheduleGenerator(mockScheduledDateGenerator, emiCalculator);
-        when(mockScheduledDateGenerator.generateRepaymentPeriods(any(), any(), any())).thenReturn(expectedRepaymentPeriods);
+        when(mockScheduledDateGenerator.generateRepaymentPeriods(any(), any(), any(), any())).thenReturn(expectedRepaymentPeriods);
```
Comment (adamsaghy) [inline fineract-progressive-loan/src/test/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleGeneratorTest.java]: Please dont do any mocks... we should be able to generate the loan schedule without any mocking!

File: fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java [hunk 4]
```diff
@@ -18,27 +18,21 @@
...
-import jakarta.persistence.Column;
-import jakarta.persistence.Embeddable;
 import java.math.BigDecimal;
+import java.math.MathContext;
 import java.math.RoundingMode;
 import java.util.Iterator;
+import lombok.Getter;
 import org.apache.fineract.organisation.monetary.data.CurrencyData;
 
-@Embeddable
```
Comment (galovics) [inline fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java:28]: Isn't this going to be a breaking change? Is there any JPA entity using this class as an embeddable?

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java (hunk 1 above)
Comment (kulminsky) [reply fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java]: Fixed.

File: fineract-progressive-loan/src/test/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleGeneratorTest.java (hunk 3 above)
Comment (kulminsky) [reply fineract-progressive-loan/src/test/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleGeneratorTest.java]: Fixed.

File: fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java (hunk 4 above)
Comment (kulminsky) [reply fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java:28]: Fixed.

File: fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java (hunk 2 above)
Comment (kulminsky) [reply fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/calc/ProgressiveEMICalculator.java]: Fixed.

File: fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java (hunk 4 above)
Comment (adamsaghy) [reply fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java:28]: I dont think we are using it in any JPA entity, but worth to double check.

File: fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java [hunk 5]
```diff
@@ -40,6 +43,16 @@ public class Money implements Comparable<Money> {
     @Column(name = "amount", scale = 6, precision = 19)
     private BigDecimal amount;
 
+    private final MathContext mc;
+
+    protected Money() {
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java:48]: It is not used. can be deleted

File: fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java (hunk 4 above)
Comment (galovics) [reply fineract-core/src/main/java/org/apache/fineract/organisation/monetary/domain/Money.java:28]: I mean guys, I'm okay with removing the JPA annotations if it is not used anywhere.

**Pull request merged**